### PR TITLE
[chore] simplified connection options by removing URL as really we just handle hostports

### PR DESCRIPTION
**Pull request by aricart**
- [BREAKING] removed `url` connection option. Instead, provide `servers` - this option can be a string or an array of strings.

### CODE REVIEW

File: workflows/test.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ jobs:
 
       - name: Set NATS Server Version
         run: echo '::set-env name=NATS_VERSION::v2.1.7'
-      - name: Get nats-server
+      - name: Get nats-servers
```
Comment (kozlovic) [inline workflows/test.yml:34]: Is that a rename-all kind of things? Not sure why you would have change nats-server to nats-servers here :-)

Comment (aricart) [reply workflows/test.yml:34]: weird my local version doesn't have it. - I will fix the title on merge.

**Comment by haywirez** [comment]
Upgrading an older brownfield project and running into connection issues — do I read this correctly, support for `nats://lollipop:SOME_PASSWORD@dokku-nats-lollipop:4222/lollipop` style connection URLs has been dropped in favor of hostPorts and credentials provided separately? Is there a parsing utility function available? Asking as projects with plugins such as https://github.com/dokku/dokku-nats still do connection URLs as the default (paging @josegonzalez)

**Comment by josegonzalez** [comment]
Providing a DSN is probably the most standard way to provide connection information and the Dokku project will not be offering any official alternative for plugins it maintains. I would recommend wrapping `nats.deno` with something that parses the DSN and sets up the connection internally if the official library won't support it.

**Comment by haywirez** [comment]
In case someone finds this issue, a naive 8 liner:

```ts
const parseNatsUrl = (url) => {
  if (url.indexOf('nats://') === -1) {
    return { servers: url }
  }
  const [usernamePassword, hostPort] = url.replace('nats://', '').split('@')
  const [user, pass] = usernamePassword.split(':')
  return { user, pass, servers: hostPort }
}
```

It's probably wiser to use something like [@soluble/dsn-parser](https://www.npmjs.com/package/@soluble/dsn-parser)

**Pull request merged**